app/controllers/works_controller.rb

```diff
@@ -0,0 +1,84 @@
+class WorksController < ApplicationController
+  def index
+    @works = Work.all
+    @movies = Work.sort_by_category("movie")
+    @books = Work.sort_by_category("book")
+    @albums = Work.sort_by_category("album")
+  end
+
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+
+    if @work.nil?
+      flash[:warning] = "Invalid Media."
+      redirect_to works_path
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = "Media added successfully"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash[:warning] = "Oops! Something is wrong. Cannot add Media."
+      redirect_to works_path
+      return
```

Comment:
I suggest:

1.  Provide the validation errors in the flash notices.
1.  Render `new` so they still get the form with the fields they'd filled out

```suggestion
      flash.now[:warning] = "Oops! Something is wrong. Cannot add Media."
      flash.now[:error] = @work.errors.messages
      render :new, status: :bad_request
      return
```